pkgs/applications/networking/browsers/nyxt/default.nix

```diff
@@ -12,36 +22,62 @@ stdenv.mkDerivation rec {
 
   src = lispPackages.nyxt;
 
-  nativeBuildInputs = [ makeWrapper wrapGAppsHook ];
-  gstBuildInputs = with gst_all_1; [
-    gstreamer gst-libav
+  nativeBuildInputs = [ wrapGAppsHook ];
+  buildInputs = [
+    cairo
+    enchant
+    gdk-pixbuf
+    glib
+    glib-networking
+    gobject-introspection
+    gsettings-desktop-schemas
+    gtk3
+    mime-types
+    pango
+  ] ++ (with gst_all_1; [
+    gstreamer
+    gst-libav
     gst-plugins-base
     gst-plugins-good
     gst-plugins-bad
     gst-plugins-ugly
-  ];
-  buildInputs = [
-    glib gdk-pixbuf cairo
-    mime-types pango gtk3
-    glib-networking gsettings-desktop-schemas
-    xclip notify-osd enchant
-  ] ++ gstBuildInputs;
+  ]);
 
-  GST_PLUGIN_SYSTEM_PATH_1_0 = lib.concatMapStringsSep ":" (p: "${p}/lib/gstreamer-1.0") gstBuildInputs;
+  binPath = lib.optionals (!stdenv.isDarwin) [ xclip ];
 
+  dontBuild = true;
   dontWrapGApps = true;
-  installPhase = ''
-    mkdir -p $out/share/applications/
-    sed "s/VERSION/$version/" $src/lib/common-lisp/nyxt/assets/nyxt.desktop > $out/share/applications/nyxt.desktop
-    for i in 16 32 128 256 512; do
-      mkdir -p "$out/share/icons/hicolor/''${i}x''${i}/apps/"
-      cp -f $src/lib/common-lisp/nyxt/assets/nyxt_''${i}x''${i}.png "$out/share/icons/hicolor/''${i}x''${i}/apps/nyxt.png"
-    done
-
-    mkdir -p $out/bin && makeWrapper $src/bin/nyxt $out/bin/nyxt \
```

Comment:
This didn't work because `gappsWrapperArgs` is computed at `preFixup`, which happens after `installPhase`.